xbmc/settings/DisplaySettings.cpp

```diff
@@ -288,14 +288,30 @@ bool CDisplaySettings::OnSettingChanging(std::shared_ptr<const CSetting> setting
     SetCurrentResolution(newRes, false);
     g_graphicsContext.SetVideoResolution(newRes);
 
+    if (m_resolutionChangeInProgress)
+    {
+      // Do not recurse into showing dialogs
+      return true;
+    }
+
     // check if the old or the new resolution was/is windowed
     // in which case we don't show any prompt to the user
     if (oldRes != RES_WINDOW && newRes != RES_WINDOW && oldRes != newRes)
     {
       if (!m_resolutionChangeAborted)
       {
-        if (HELPERS::ShowYesNoDialogText(CVariant{13110}, CVariant{13111}, CVariant{""}, CVariant{""}, 15000) !=
-          DialogResponse::YES)
+        m_resolutionChangeInProgress = true;
+#if defined(HAVE_WAYLAND)
```

Comment:
I think here at the latest you noticed that DisplaySettings is a can of worms :)